onnxruntime/core/providers/dnnl/subgraph/dnnl_dequantizelinear.cc

```diff
@@ -113,6 +123,20 @@ void DnnlDequantizeLinear::CreatePrimitive(DnnlSubgraphPrimitive& sp, DnnlNode&
   }
 }
 
+bool DnnlDequantizeLinear::isZeroPointNonZero(dnnl::memory* zp_mem) {
+  // Because zp will always be int8, uint8 or int32, this cast is always valid
```

Comment:
is it always valid?  Isn't int32 4 bytes while int8 and uint8 are 1 byte? are int8 and uint8 padded to be the same size as int32? Always being able to cast them to `int*` feels like its missing something.

Reply:
Agree I'll change the cast to int8* and just adjust the number of checks by x4 when the original datatype is int32